docs/reference/analysis.asciidoc

```diff
@@ -60,3 +61,5 @@ include::analysis/tokenfilters.asciidoc[]
 include::analysis/charfilters.asciidoc[]
 
 include::analysis/normalizers.asciidoc[]
+
+include::analysis/synonyms.asciidoc[]
```

Comment:
This whole `synonyms.asciidoc` file should not be here, as it is more like a guide rather than documentation, as it doesn't contain any original information.

Let's put it where we have our guides user "Search your data" (https://www.elastic.co/guide/en/elasticsearch/reference/8.9/search-your-data.html) section.